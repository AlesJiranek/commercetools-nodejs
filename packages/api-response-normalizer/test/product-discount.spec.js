import normalizeProductDiscount from '../src/product-discount'
import productDiscountsResponse from './fixtures/product-discount.json'

describe('when the response is from `/product-type/:id`', () => {
  it('should normalize response', () => {
    expect(normalizeProductDiscount(productDiscountsResponse)).toMatchSnapshot()
  })
})
