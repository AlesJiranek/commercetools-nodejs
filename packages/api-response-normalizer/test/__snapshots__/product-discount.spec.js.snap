// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the response is from \`/product-type/:id\` should normalize response 1`] = `
Object {
  "entities": Object {
    "productDiscounts": Object {
      "00801b20-b114-4ef5-9382-5082f1be834f": Object {
        "attributeTypes": Object {},
        "createdAt": "2017-01-12T10:41:06.441Z",
        "description": Object {
          "en": "<p>1 Euro Discount on chicken!</p>",
        },
        "id": "00801b20-b114-4ef5-9382-5082f1be834f",
        "isActive": false,
        "lastModifiedAt": "2017-02-28T10:48:17.503Z",
        "name": Object {
          "en": "1 Euro Discount",
        },
        "predicate": "product.id = \\"6490b586-6576-4429-b75f-95f0c3b1ab2f\\" and variant.id = 1",
        "references": Array [
          Object {
            "id": "6490b586-6576-4429-b75f-95f0c3b1ab2f",
            "typeId": "product",
          },
        ],
        "sortOrder": "0.8",
        "value": Object {
          "money": Array [
            Object {
              "centAmount": 112,
              "currencyCode": "EUR",
            },
          ],
          "type": "absolute",
        },
        "version": 8,
      },
    },
  },
  "result": "00801b20-b114-4ef5-9382-5082f1be834f",
}
`;
